### PR TITLE
Switch to using 'authority' instead of 'host' when building parent URI in PinotFS::move

### PR DESCRIPTION
**Pull request by cbalci**
## Description
Switch to using 'authority' instead of 'host' when building parent URI in PinotFS:move method.
This fixes a bug when a FileSystem implementation uses URI with ports. Using 'host' strips the port information from the URI as:
```
URI uri = new URI("hdfs://localhost:1234/my/path");

uri.getHost(); // "localhost"
uri.getAuthority(); // "localhost:1234"
```

- Also added a test case which captures the failure.

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)

### CODE REVIEW

**Comment by cbalci** [comment]
@Jackie-Jiang @chenboat @elonazoulay ptal

File: pinot-spi/src/test/java/org/apache/pinot/spi/filesystem/PinotFSTest.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class PinotFSTest {
+  public String srcForMoveFile = "myfs://myhost1:1234/root/file1";
+  public String dstForMoveFile = "myfs://myhost2:1234/someDir/file1";
+
+  @Test
+  public void testMoveFileUriGeneration() throws Exception {
```
Comment (chenboat) [inline pinot-spi/src/test/java/org/apache/pinot/spi/filesystem/PinotFSTest.java:39]: can you add a test where there is no port in the authority?

Comment (cbalci) [reply pinot-spi/src/test/java/org/apache/pinot/spi/filesystem/PinotFSTest.java:41]: Added.

**Review by Jackie-Jiang — APPROVED** [review]
LGTM. Let's wait for @snleee and @elonazoulay to take a look before merging

**Comment by cbalci** [comment]
@snleee @elonazoulay please take a look when you get a chance.

**Review by chenboat — APPROVED** [review]

**Comment by cbalci** [comment]
@Jackie-Jiang @chenboat thanks for your reviews. Haven't been able to reach @snleee and @elonazoulay. Can we move forward merging this?

**Pull request merged**